appsec-kit-v24/src/main/java/com/vaadin/appsec/v24/views/AppSecView.java

```diff
@@ -126,15 +131,7 @@ public void onAttach(AttachEvent event) {
         super.onAttach(event);
         removeScanListener();
         scanListener = AppSecService.getInstance()
-                .addScanEventListener(scanEvent -> AppSecView.this.getUI()
-                        .ifPresent(ui -> ui.access(() -> {
-                            scanNowButton.setEnabled(true);
-                            refresh();
-                            if (PushMode.MANUAL == ui.getPushConfiguration()
-                                    .getPushMode()) {
-                                ui.push();
-                            }
-                        })));
+                .addScanEventListener(createScanEventListener());
```

Comment:
Let's make `createScanEventLister()` to _be_ the listener instead to return one, i.e.

```java
    .addScanEventListener(this::handleScanEvent);
}

private void handleScanEvent(AppSecScanEvent event) {
    getUI().ifPresent(this::handleScanNowButton);
}
```

Reply:
Updated.